src/Tests/WebApp.fs

```diff
@@ -2075,4 +2075,32 @@ let tests =
                 Expect.equal slot.Name.Value "webapp/deployment" "Slot name was not as expected"
                 Expect.equal slot.AlwaysOn false "Slot was not set to Always On -> false"
             }
+            test "WebApp supports runtime on slot" {
+                let slot =
+                    appSlot {
+                        name "warm-up"
+                        runtime Runtime.DotNet80
+                    }
+
+                let app =
+                    webApp {
+                        name "webapp"
+                        add_slot slot
+                    }
+
+                Expect.isTrue (app.CommonWebConfig.Slots.ContainsKey "warm-up") "Config should contain slot"
+
+                let slots =
+                    app
+                    |> getResources
+                    |> getResource<Arm.Web.Site>
+                    |> List.filter (fun x -> x.ResourceType = Arm.Web.slots)
+                // Default "production" slot is not included as it is created automatically in Azure
+                Expect.hasLength slots 1 "Should only be 1 slot"
+
+                let connStrings =
```

Comment:
**Suggestion**: this should probably be named something like `netFrameworkVersion`
```suggestion
                let netFrameworkVersion=
```